lib/BleFingerprint/BleFingerprint.h

```diff
@@ -68,7 +68,7 @@ class BleFingerprint
 
     bool report(JsonObject *doc);
 
-    bool query();
+    bool query(const char* roomsTopic,AsyncMqttClient* mqttClient);
```

Comment:
Seems like we're doing something wrong if we need to directly send to mqtt in here.  Can we instead put the data somewhere, and on the outside check for changes and send to mqtt where we normally do?  What if mqtt is momentarily down?  This will most likely cause trouble...

Reply:
We could save the messages we want to send and let the report task handle the rest, therefore creating some kind of message queue would be great and the report task will just work on it. This kind of queue should be threadsafe, right? But we would have to return the messages from the Blefingerprint::query function with some kind of touple. And then in main add it to the message queue. Would you be fine with this kind of attempt? Also thanks for the quick replies on my buggy code.

Reply:
It could just be a newed up object added to fingerprint, right?  Something like extendedInfo of type Miflora?  We could use that to add BTHome support as well with a different class with the same base clasee.

Reply:
So that the report task checks the fingerprint object for new mqtt messages

Reply:
I'd make the object have the properties, and have an send to mqtt function that would be able to send it.  It should have some smarts about not reporting if stuff didn't change...

Reply:
Added something like that, if you want to take a look

Reply:
@DTTerastar Maybe i could add even add the Handling for the BLE Scan to that class structure two,  but would it kind of take the simplicity out of that aproach?